libs/wingsdk/src/http/http.ts

```diff
@@ -271,6 +327,61 @@ export class Util {
     });
   }
 
+  /**
+   * Parses the input URL String using WHATWG URL API and returns an URL Struct.
+   * @param urlString The URL String to be parsed.
+   * @throws Will throw an error if the input String is not a valid URL.
+   * @inflight
+   * @returns An URL Struct.
+   */
+  public static parseUrl(urlString: string): Url {
+    try {
+      const nodeUrl = new NodeUrl(urlString);
+      return {
+        href: nodeUrl.href,
+        protocol: nodeUrl.protocol,
+        host: nodeUrl.host,
+        hostname: nodeUrl.hostname,
+        port: nodeUrl.port,
+        pathname: nodeUrl.pathname,
+        search: nodeUrl.search,
+        hash: nodeUrl.hash,
+        origin: nodeUrl.origin,
+        username: nodeUrl.username,
+        password: nodeUrl.password,
+      };
+    } catch (error) {
+      throw new Error(`Invalid URL: ${urlString}`);
+    }
+  }
+
+  /**
+   * Serializes an URL Struct to a String.
+   * @param url The URL Struct to be formatted.
+   * @throws Will throw an error if the input URL has invalid fields.
+   * @inflight
+   * @returns A formatted URL String.
+   */
+  public static formatUrl(url: Url, options?: FormatUrlOptions): string {
+    try {
+      const nodeUrl = new NodeUrl(url.href);
+      return format(nodeUrl, {
+        auth: options?.auth,
+        fragment: options?.fragment,
+        search: options?.search,
+        unicode: options?.unicode,
```

Comment:
```suggestion
      return format(nodeUrl, options);
```

Reply:
This change was [already made](https://github.com/winglang/wing/pull/4329/commits/78083d67db0a28200976580a037a7e46b5bbecaf) 2 days ago.

Reply:
Cool! thanks for letting me know :) I'll test it on aws and approve if everything passes :)